src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultControllerModelBuilder.cs

```diff
@@ -80,6 +85,22 @@ protected virtual ControllerModel CreateControllerModel([NotNull] TypeInfo typeI
             AddRange(controllerModel.Filters, attributes.OfType<IFilter>());
             AddRange(controllerModel.RouteConstraints, attributes.OfType<IRouteConstraintProvider>());
 
+            var enableCors = attributes.OfType<IEnableCorsMetadata>().SingleOrDefault();
```

Comment:
THinking a bit more, for such a scenario it would probably make more sense to do it using the CorsMiddleware instead of registering the filter. I will add the test when middleware is in.